src/syntax/parser.rs

```diff
@@ -1086,6 +1113,41 @@ fn validate_args(p: &mut Parser, m: Marker) {
     }
 }
 
+fn validate_unpacking(p: &mut Parser, m: Marker) {
+    let mut used_spread = false;
+    let mut used = HashSet::new();
+    for child in p.post_process(m) {
+        match child.kind() {
+            SyntaxKind::Ident => {
+                if !(child.text() == "_") && !used.insert(child.text().clone()) {
+                    child.convert_to_error("duplicate identifier");
+                }
+            }
+            SyntaxKind::Spread => {
+                let Some(within) = child.children_mut().last_mut() else { continue };
+                if within.kind() != SyntaxKind::Ident {
+                    within.convert_to_error(eco_format!(
+                        "expected identifier, found {}",
+                        within.kind().name(),
+                    ));
+                    child.make_erroneous();
+                }
+                if used_spread {
+                    child.convert_to_error("multiple argument sinks");
```

Comment:
```suggestion
                    child.convert_to_error("multiple destructuring sinks");
```
maybe reword to e.g. "at most one destructuring sink is allowed". if args has the same message, reword that, too.